system-contracts/test/L1Messenger.spec.ts

```diff
@@ -0,0 +1,517 @@
+    });
+
+    it("should revert state diff compression version mismatch", async () => {
+      await (
+        await l1Messenger
+          .connect(knownCodeStorageAccount)
+          .requestBytecodeL1Publication(await ethers.utils.hexlify(utils.hashBytecode(bytecodeData.content)), {
+            gasLimit: 130000000,
+          })
+      ).wait();
+      emulator.messages[0] = { lengthBytes: logData.currentMessageLengthBytes, content: logData.message };
+      // modify version to trigger the revert
+      emulator.version = ethers.utils.hexZeroPad(ethers.utils.hexlify(66), 1);
+      await expect(
+        l1Messenger
+          .connect(bootloaderAccount)
+          .publishPubdataAndClearState(emulator.buildTotalL2ToL1PubdataAndStateDiffs())
+      ).to.be.rejectedWith("state diff compression version mismatch");
+    });
+
+    it("should revert extra data", async () => {
+      emulator.version = ethers.utils.hexZeroPad(ethers.utils.hexlify(1), 1);
+      // add extra data to trigger the revert
+      await expect(
+        l1Messenger
+          .connect(bootloaderAccount)
+          .publishPubdataAndClearState(
+            ethers.utils.concat([emulator.buildTotalL2ToL1PubdataAndStateDiffs(), Buffer.alloc(1, 64)])
+          )
+      ).to.be.rejectedWith("Extra data in the totalL2ToL1Pubdata array");
+    });
+  });
+
+  describe("sendL2ToL1Log", async () => {
+    it("should revert when not called by the system contract", async () => {
+      await expect(l1Messenger.sendL2ToL1Log(logData.isService, logData.key, logData.value)).to.be.rejectedWith(
+        "This method require the caller to be system contract"
+      );
+    });
+
+    it("should emit L2ToL1LogSent event when called by the system contract", async () => {
+      await expect(
+        l1Messenger
+          .connect(l1MessengerAccount)
+          .sendL2ToL1Log(logData.isService, ethers.utils.hexlify(logData.key), ethers.utils.hexlify(logData.value))
+      )
+        .to.emit(l1Messenger, "L2ToL1LogSent")
+        .withArgs([
+          0,
+          logData.isService,
+          1,
+          l1MessengerAccount.address,
+          ethers.utils.hexlify(logData.key),
+          ethers.utils.hexlify(logData.value),
+        ]);
+    });
+
+    it("should emit L2ToL1LogSent event when called by the system contract with isService false", async () => {
+      await expect(
+        l1Messenger
+          .connect(l1MessengerAccount)
+          .sendL2ToL1Log(false, ethers.utils.hexlify(logData.key), ethers.utils.hexlify(logData.value))
+      )
+        .to.emit(l1Messenger, "L2ToL1LogSent")
+        .withArgs([
+          0,
+          false,
+          1,
+          l1MessengerAccount.address,
+          ethers.utils.hexlify(logData.key),
+          ethers.utils.hexlify(logData.value),
+        ]);
+    });
+
+    it("should revert when called by the system contract with empty key & value", async () => {
```

Comment:
Do you think that tests:

`should revert when called by the system contract with empty key & value`, `should revert when called by the system contract with key & value > 32 bytes`, `should revert when called by the system contract with key & value < 32 bytes` are useful? 

They basically test that Solidity abi decoding works correctly. These arent edge cases for ABI decoding, so probably it doesn't make sense to test the Solidity compiler

Reply:
fixed